src/main/java/software/amazon/event/kafkaconnector/EventBridgeWriter.java

```diff
@@ -144,10 +144,14 @@ public List<EventBridgeResult<EventBridgeEventId>> putItems(List<SinkRecord> rec
   private Stream<EventBridgeResult<EventBridgeEventId>> sendToEventBridge(
       List<MappedSinkRecord<PutEventsRequestEntry>> items) {
     try {
-      var request =
+      var requestBuilder =
           PutEventsRequest.builder()
-              .entries(items.stream().map(MappedSinkRecord::getValue).collect(toList()))
-              .build();
+              .entries(items.stream().map(MappedSinkRecord::getValue).collect(toList()));
+
+      if (config.endpointID != null && !config.endpointID.isEmpty()) {
+        requestBuilder.endpointId(config.endpointID);
+      }
```

Comment:
no test for this?

Reply:
I can check if we have some tests already here, mocking in Java is not my expertise 😀 

I only test in prod xD

Reply:
Added